File: Android/app/src/test/java/com/lunesu/bcferries/FerryPierTest.kt
package com.lunesu.bcferries

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class FerryPierTest {
    @Test
    fun testName() {
        assertEquals("SwartzBay", FerryPier.SwartzBay.name)
    }

    @Test
    fun testToString() {
        assertEquals("SwartzBay", FerryPier.SwartzBay.toString())
    }

    @Test
    fun testValueOf() {
        assertEquals(FerryPier.SwartzBay, FerryPier.valueOf("SwartzBay"))
    }

    @Test
    fun testFromCode() {
        assertEquals(FerryPier.Tsawwassen, FerryPier.fromCode("TSA"))
        assertEquals(FerryPier.DepartureBay, FerryPier.fromCode("NAN"))
        assertNull(FerryPier.fromCode("SGI"))
    }

    @Test
    fun testMainland() {
        assertEquals(true, FerryPier.Tsawwassen.isMainland)
        assertEquals(true, FerryPier.HorseshoeBay.isMainland)
        assertEquals(false, FerryPier.SwartzBay.isMainland)
        assertEquals(false, FerryPier.BowenIsland.isMainland)
    }

    @Test
    fun testNearest() {
        assertEquals(FerryPier.Tsawwassen, FerryPier.findNearest(49.007, -123.129))
        assertEquals(FerryPier.SwartzBay, FerryPier.findNearest(48.689, -123.410))
        assertEquals(FerryPier.BowenIsland, FerryPier.findNearest(49.381, -123.331))
    }

    @Test
    fun testNearestCustom() {
        assertEquals(FerryPier.Langdale, FerryPier.findNearest(49.007, -123.129, arrayOf(FerryPier.Langdale)))
    }
}

File: iOS/BCFerriesTests/FerryPierTest.swift
//
//  FerryPierTest.swift
//  BCFerriesTests
//

import XCTest
@testable import BCFerries

class FerryPierTest: XCTestCase {

    func testName() {
        XCTAssertEqual("SwartzBay", FerryPier.SwartzBay.rawValue)
    }

    func testToString() {
        XCTAssertEqual("SwartzBay", FerryPier.SwartzBay.toString())
    }

    func testValueOf() {
        XCTAssertEqual(FerryPier.SwartzBay, FerryPier.valueOf("SwartzBay"))
    }

    func testCode() {
        XCTAssertEqual(FerryPier.Tsawwassen, FerryPier.fromCode("TSA"))
        XCTAssertEqual(FerryPier.DepartureBay, FerryPier.fromCode("NAN"))
        XCTAssertNil(FerryPier.fromCode("SGI"))
        for pier in FerryPier.ENUMS {
            XCTAssertEqual(pier, FerryPier.fromCode(pier.code))
        }
    }

    func testMainland() {
        XCTAssertTrue(FerryPier.Tsawwassen.isMainland)
        XCTAssertTrue(FerryPier.HorseshoeBay.isMainland)
        XCTAssertFalse(FerryPier.SwartzBay.isMainland)
        XCTAssertFalse(FerryPier.BowenIsland.isMainland)
    }

    func testNearest() {
        XCTAssertEqual(FerryPier.Tsawwassen, FerryPier.findNearest(latitude: 49.007, longitude: -123.129))
        XCTAssertEqual(FerryPier.SwartzBay, FerryPier.findNearest(latitude: 48.689, longitude: -123.410))
        XCTAssertEqual(FerryPier.BowenIsland, FerryPier.findNearest(latitude: 49.381, longitude: -123.331))
    }

    func testNearestCustom() {
        XCTAssertEqual(FerryPier.Langdale, FerryPier.findNearest(latitude: 49.007, longitude: -123.129, piers: [.Langdale]))
    }
}
